### ISSUE TITLE
rustc build warning: diagnostic code E0005 already used

### ISSUE THREAD

**Issue by mitaa** [rust-lang/rust#27279]
Building the master branch yields this warning. 

```
src/librustc/middle/check_match.rs:1050:5: 1059:6 warning: diagnostic code E0005 already used
src/librustc/middle/check_match.rs:1050     for input in &decl.inputs {
src/librustc/middle/check_match.rs:1051         is_refutable(cx, &*input.pat, |pat| {
src/librustc/middle/check_match.rs:1052             span_err!(cx.tcx.sess, input.pat.span, E0005,
src/librustc/middle/check_match.rs:1053                 "refutable pattern in function argument: `{}` not covered",
src/librustc/middle/check_match.rs:1054                 pat_to_string(pat)
src/librustc/middle/check_match.rs:1055             );
                                        ...
src/librustc/middle/check_match.rs:1025:66: 1030:6 note: previous invocation
src/librustc/middle/check_match.rs:1025     is_refutable(cx, &*static_inliner.fold_pat(loc.pat.clone()), |pat| {
src/librustc/middle/check_match.rs:1026         span_err!(cx.tcx.sess, loc.pat.span, E0005,
src/librustc/middle/check_match.rs:1027             "refutable pattern in {} binding: `{}` not covered",
src/librustc/middle/check_match.rs:1028             name, pat_to_string(pat)
src/librustc/middle/check_match.rs:1029         );
src/librustc/middle/check_match.rs:1030     });
```

It seems like it should be easy to solve this using the same approach as in #27188
